frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/PushCurrentBranchAction.java

```diff
@@ -10,4 +16,20 @@
   public Option<String> getNameOfBranchUnderAction(AnActionEvent anActionEvent) {
     return getCurrentBranchNameIfManaged(anActionEvent);
   }
+
+  @Override
+  @UIEffect
+  public void onUpdate(AnActionEvent anActionEvent) {
+    var isEnabledAndVisible = getNameOfBranchUnderAction(anActionEvent)
+        .flatMap(bn -> getGitMacheteBranchByName(anActionEvent, bn))
+        .map(b -> b.getSyncToRemoteStatus().getRelation())
+        .map(strs -> List.of(AheadOfRemote, DivergedFromAndNewerThanRemote, Untracked).contains(strs))
```

Comment:
Drop a comment why it isn't equal to `com.virtuslab.gitmachete.frontend.actions.base.BasePushBranchAction#getEligibleRelations`... my reasoning is that we want to make `DivergedFromAndOlderThanRemote` only available to those who really really want it (hence context menu) but not suggested by default (hence not toolbar)

Reply:
Similar comment to other actions: either use some shared method that provides the eligible statuses, or specify why the list of eligible statuses differs from what's in the context-menu counterpart

Reply:
\+ in either case, don't define `getEligibleStatuses` in the base class if it only accurately matches one of the subclasses and not the other

Reply:
The deeper `getEligibleRelations` decides about "enability" of an action.
The higher logic (here) decides about visibility.

Note that: visibility =/= enability

`getEligibleRelations` matches both (for current and selected) implementations in scope of "enability". 
The special handling of visibility (happening here) is required. 

Keep in mind that even though we do not want to show some actions within the toolbar they should be still enabled (since they may be called from `Find action`).

Reply:
Okay! pls document that in `getEligibleRelations` since I didn't think of this distinction (most likely, I just silently assumed that they correspond to **enabled&&visible** actions, not just **enabled**).

Reply:
Done